### ISSUE TITLE
fixef when varying slope is integer gives error:  REAL() can only be applied to a 'numeric', not a 'integer'

### ISSUE THREAD

**Issue by MatthieuStigler** [lrberge/fixest#32]
Merci Laurent pour ce super paquet!!

It seems `fixef()` has a trouble accommodating integer variables when using varying slopes, i.e. if `x3` is integer in `species[x3]`?

See:

``` r
library(fixest)

base_vs = iris 
names(base_vs) = c(paste0("x", 1:4), "species")
base_vs$x5 <- as.integer(round(base_vs$x3))

out_double <- feols(x1 ~ x2 | species[x3], base_vs)
out_integer <- feols(x1 ~ x2 | species[x5], base_vs)

out <- fixef(out_double)
fixef(out_integer)
#> Error in cpp_demean(y = S, X_raw = 0, r_weights = 0, iterMax = 1000L, : REAL() can only be applied to a 'numeric', not a 'integer'
```

<sup>Created on 2020-07-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by lrberge** [comment]
Salut Matthieu, mais tout le plaisir est pour moi ! :-)

It's a bug that's been corrected in version 0.5.0 (#20). 

Now it gives:
```R
summary(fixef(out_integer))
#> Fixed-effects/Slope coefficients
#>                                species x5 (slopes: species)
#> Number of fixed-effects/slopes       3                    3
#> Number of references                 0                    0
#> Mean                              2.19                0.427
#> Variance                         0.623               0.0766
#> 
#> COEFFICIENTS:
#>   species: virginica versicolor setosa
#>                1.393      2.193  2.972
#> -----
#>   x5 (slopes: species): virginica versicolor setosa
#>                            0.6368     0.5303  0.113
```

**Comment by MatthieuStigler** [comment]
oh wow, I was late in my version updates, sorry about that!
